src/pages/transactions/index.tsx

```diff
@@ -96,6 +94,15 @@ export const requestTransactionsDefault = async (
           ) {
             assets.push(contract.parameter.currencyID);
           }
+
```

Comment:
the loop is doing too much things, should be abstracted to smaller functions handling each case individually